src/client/search-init.ts

```diff
@@ -8,6 +8,10 @@ const input = container.querySelector<HTMLInputElement>("input")!;
 const load = () => import("observablehq:search");
 input.addEventListener("focus", load, {once: true});
 input.addEventListener("keydown", load, {once: true});
+input.addEventListener("focus", () => {
+  input.value = sessionStorage.getItem("search-query") ?? "";
+  input.select();
```

Comment:
```suggestion
  if (input.value) return; // only restore the query if empty
  input.value = sessionStorage.getItem("search-query") ?? "";
  input.select();
```

Calling `input.select` on focus interferes with the normal behavior, _e.g._, you can’t click to place the caret or select a subset of the text because all the text is automatically selected. I think we would only want to assign the value and select it if the input is empty.